hydroflow_lang/src/graph/ops/flatten.rs

```diff
@@ -25,6 +28,11 @@ pub const FLATTEN: OperatorConstraints = OperatorConstraints {
     is_external_input: false,
     ports_inn: None,
     ports_out: None,
+    properties: FlowProperties {
+        deterministic: FlowPropertyVal::CodeBlock,
```

Comment:
Seems like `flatten` should be `Preserve` and `Preserve` -- it doesn't have an associate closure. Does it?

Reply:
Yup that seems right